classes/ror/Repository.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * @file classes/ror/Repository.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @brief A repository to find and manage rors.
+ */
+
+namespace PKP\ror;
+
+use APP\core\Request;
+use Illuminate\Support\Facades\App;
+use PKP\plugins\Hook;
+use PKP\services\PKPSchemaService;
+
+class Repository
+{
+    public DAO $dao;
+
+    /** @var string $schemaMap The name of the class to map this entity to its schema */
+    public $schemaMap = maps\Schema::class;
+
+    /** @var Request */
+    protected $request;
+
+    /** @var PKPSchemaService<Ror> */
+    protected $schemaService;
+
+    public function __construct(DAO $dao, Request $request, PKPSchemaService $schemaService)
+    {
+        $this->dao = $dao;
+        $this->request = $request;
+        $this->schemaService = $schemaService;
+    }
+
+    /** @copydoc DAO::newDataObject() */
+    public function newDataObject(array $params = []): Ror
+    {
+        $object = $this->dao->newDataObject();
+        if (!empty($params)) {
+            $object->setAllData($params);
+        }
+        return $object;
+    }
+
+    /** @copydoc DAO::exists() */
+    public function exists(int $id): bool
+    {
+        return $this->dao->exists($id);
+    }
+
+    /** @copydoc DAO::get() */
+    public function get(int $id): ?Ror
+    {
+        return $this->dao->get($id);
+    }
+
+    /** @copydoc DAO::getCollector() */
+    public function getCollector(): Collector
+    {
+        return App::make(Collector::class);
+    }
+
+    /**
+     * Get an instance of the map class for mapping rors to their schema.
+     */
+    public function getSchemaMap(): maps\Schema
+    {
+        return app('maps')->withExtensions($this->schemaMap);
+    }
+
+    /** @copydoc DAO::insert() */
+    public function add(Ror $ror): int
+    {
+        $id = $this->dao->insert($ror);
+        Hook::call('Ror::add', [$ror]);
```

Comment:
Before the hook I think we need to get the ror (by ID) again, because else the code using the hook will not bet the ror ID